std-bits/base/src/main/java/org/enso/base/enso_cloud/HideableValue.java

```diff
@@ -0,0 +1,33 @@
+package org.enso.base.enso_cloud;
+
+/** Represents a value that is input of various operation that may contain a Secret. */
+public sealed interface HideableValue permits HideableValue.SecretValue, HideableValue.PlainValue {
+
+  record SecretValue(String secretId) implements HideableValue {
+    @Override
+    public String render() {
+      return "__SECRET__";
+    }
+
+    @Override
+    public String safeResolve() throws EnsoSecretAccessDenied {
+      throw new EnsoSecretAccessDenied();
+    }
+  }
+
+  record PlainValue(String value) implements HideableValue {
+    @Override
+    public String render() {
+      return value;
+    }
+
+    @Override
+    public String safeResolve() throws EnsoSecretAccessDenied {
+      return value;
+    }
+  }
+
+  String render();
```

Comment:
This is usually called `toString()` in Java.

Reply:
Fair, I wanted to make sure we call it explicitly to distinguish the usages where we convert secrets to `__SECRET__` vs usages where we disallow secrets vs usages where we read actual secret values.

I can change to `toString` if you think that does not clash with the above.